src/api/github-client.ts

```diff
@@ -0,0 +1,239 @@
+      return this.getTagInfoByVersion(CHE_OPERATOR_REPO, version)
+    } else if (installer === 'helm') {
+      return this.getTagInfoByVersion(CHE_REPO, version)
+    }
+    throw new Error(`Unsupported installer: ${installer}`)
+  }
+
+  /**
+   * Gets last 50 tags from the given repository.
+   * @param repo repository name to list tag in
+   * @param prefix return only tags that starts with given prefix
+   */
+  private async listLatestTags(repo: string, prefix = ''): Promise<TagInfo[]> {
+    let response = await this.octokit.repos.listTags({ owner: OWNER, repo, per_page: 50 })
+    const tags = response.data
+    if (prefix) {
+      return tags.filter(tag => tag.name.startsWith(prefix))
+    }
+    return tags
+  }
+
+  /**
+   * Gets tag info if it exists.
+   * @param repo repository name to search for the tag in
+   * @param tagName name of the tag
+   */
+  private async getTag(repo: string, tagName: string): Promise<TagInfo | undefined> {
+    try {
+      const tagRefResp = await this.octokit.git.getRef({ owner: OWNER, repo, ref: `tags/${tagName}` })
+      const tagRef = tagRefResp.data
+      const downloadUrlResp = await this.octokit.repos.downloadZipballArchive({ owner: OWNER, repo, ref: tagRef.object.sha })
+       // Simulate tag info
+      return {
+        name: tagName,
+        commit: {
+          sha: tagRef.object.sha,
+          url: tagRef.object.url,
+        },
+        zipball_url: downloadUrlResp.url,
+      }
+    } catch (e) {
+      if (e.status !== 404) {
+        throw e
+      }
+      // Not found, return undefined
+    }
+  }
+
+  /**
+   * Returns latest commit information in tag format.
+   * @param repo repository name to get the latest commit from
+   */
+  private async getLastCommitInfo(repo: string): Promise<TagInfo> {
+    const listCommitsResponse = await this.octokit.repos.listCommits({ owner: OWNER, repo, per_page: 1 })
+    if (listCommitsResponse.status !== 200) {
+      throw new Error(`Failed to get list of commits from the repository '${repo}'. Request: ${listCommitsResponse.url}, response: ${listCommitsResponse.status}`)
+    }
+    const lastCommit = listCommitsResponse.data[0]
+
+    const downloadZipResponse = await this.octokit.repos.downloadZipballArchive({
+      owner: OWNER,
+      repo,
+      ref: lastCommit.sha!,
+    })
+    const zipball_url = downloadZipResponse.url
+
+    // Simiulate tag info to have similar return type
+    return {
+      name: 'next',
+      commit: {
+        sha: lastCommit.sha!,
+        url: lastCommit.commit.url,
+      },
+      zipball_url,
+    }
+  }
+
+  /**
+   * Returns tag/commit information about given version.
+   * The informaton includes zip archive download link.
+   * If non-existing version is given, then undefined will be returned.
+   * @param repo repository name
+   * @param version version or version prefix. If only prefix is given, the latest one that match will be choosen.
```

Comment:
Only proposal: maybe we should use "-" after arguments name everywhere:
@param version - version or version prefix. If only prefix is given, the latest one that match will be choosen.

Reply:
Actually `-` is added automatically by LS.